### ISSUE TITLE
Unintuitive dependency injection behavior in the authorize method

### ISSUE THREAD

**Issue by mortenscheel** [lorisleiva/laravel-actions#38]
The `authorize` method support dependency injection of models resolved through route model binding.
Since this is a feature that's usually used in Controllers, I expected it to behave in a similar way, but it doesn't.

Example:
```php
public function authorize(Request $request, Customer $customer)
{
    /*
     * If the "customer" attribute isn't provided, the injected $customer parameter
     * is an empty instance of Customer (no id or any other attributes).
     */
}

public function rules()
{
    return [
        'customer' => 'required|exists:customers,id',
    ];
}
```
The missing `customer` attribute isn't caught by the validation rule, since that is checked _after_ authorization.

The behavior I expected was that since (1) the `$customer` parameter is mandatory and (2) a `Customer` couldn't be resolved, an exception should have been thrown. Had the `$customer` parameter been optional, I would have expected a `null` value, and not an empty model instance.

**Comment by lorisleiva** [comment]
This is very strange. The `ResolvesMethodDependency` trait is heavily inspired by Laravel's own Dependency resolver and, as such, the provided example should raise a `ModelNotFoundException`. See code below:

https://github.com/lorisleiva/laravel-actions/blob/1cf1b603953a72908b641e946c172908972cb531/src/Concerns/ResolvesMethodDependencies.php#L67-L74

I'm going to see if I can write a regression test for this.

**Comment by lorisleiva** [comment]
Actually, we already have a test for this:

https://github.com/lorisleiva/laravel-actions/blob/1cf1b603953a72908b641e946c172908972cb531/tests/ResolvesMethodDependenciesTest.php#L120-L131

Could you help me reproduce this bug with a fresh Laravel install?

**Comment by mortenscheel** [comment]
Sorry, my description of the problem is missing an important piece of information. The package does indeed throw a `ModelNotFoundException` when the attributes contain an invalid model id. I'm talking about a situation where the attribute is entirely missing.

I made the (faulty) assumption that since I had defined the `customer` attribute as `required`,  the validation logic would throw an error if it was missing - _before_ the `authorize` method is called. I suppose it's more a misunderstanding on my part than it's an issue with this package, so I'm going to close this issue.
